File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        output = ""
        long_word = word1 if len(word1) >= len(word2) else word2
        short_len = len(word1) if len(word1) < len(word2) else len(word2)
        for i in range(short_len):
            output += word1[i] + word2[i]

        output += long_word[short_len:]
        return output

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        char[] word1Arr = word1.toCharArray();
        char[] word2Arr = word2.toCharArray();

        String shortWord;
        if (word1.length() <= word2.length()) {
            shortWord = word1;
        } else shortWord = word2;

        String longWord;
        if (word1.length() >= word2.length()) {
            longWord = word1;
        } else longWord = word2;

        int longWordLength = Math.max(word1.length(), word2.length());
        

        StringBuilder sb = new StringBuilder();

        for (int i=0; i<shortWord.length(); ++i) {
            sb.append(word1Arr[i]);
            sb.append(word2Arr[i]);
        }

        for (int i=shortWord.length(); i<longWordLength; ++i) {
            sb.append(longWord.charAt(i));
        }

        return sb.toString();
        
    }
}